src/daemon/dbus_name_watcher.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (C) 2019 Luxoft Sweden AB
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+//
+// SPDX-License-Identifier: MPL-2.0
+
+#ifndef CONNECTIVITY_MANAGER_DAEMON_DBUS_NAME_WATCHER_H
+#define CONNECTIVITY_MANAGER_DAEMON_DBUS_NAME_WATCHER_H
+
+#include <giomm.h>
+#include <glibmm.h>
+
+#include <utility>
+
+namespace ConnectivityManager::Daemon
+{
+    // RAII helper for watching a name on the bus.
+    //
+    // Meant to be used when service should call a method in client (as is the case for e.g. the
+    // UserInputAgent interface). Service needs to know if a client disconnects from the bus before
+    // method should be called.
+    class DBusNameWatcher
```

Comment:
Seems like this could be useful for other projects as well? Maybe consider moving it to the generator or a utils lib?

Reply:
Maybe. Unsure what we will do long term with reusable code for our components. I think it makes sense to do stuff for a while and the look at what makes sense to share when we have a clearer picture. (Of course, this relies on us getting the time to refactor.)

Reply:
:+1: